nixos/modules/hardware/decklink.nix

```
@@ -0,0 +1,21 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.decklink;
+  kernelPackages = config.boot.kernelPackages;
+in
+{
+  options.hardware.decklink.enable = mkEnableOption "Enable hardware support for the Blackmagic Design Decklink audio/video interfaces.";
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "blackmagic" "blackmagic-io" "snd_blackmagic-io" ];
+    boot.extraModulePackages = [ kernelPackages.decklink ];
+    systemd.services."DecklinkVideoHelper" = {
```

```suggestion
    systemd.services.DecklinkVideoHelper = {
```

Quotes do nothing here.

---

Given this `systemd` unit is an exact copy of the unit provided by the vendor can we just use it instead? As @petabyteboy mentioned we might be able to apply some hardening options, but on top of the include `systemd` unit.

`systemd.packages = [ pkgs.blackmagicDesktopVideo ];` assuming the `.service` file is included in the package.

---

That sounds good to me. We can add a warning and link to an issue to fix it later.
Maybe upstream could even change their systemd unit to use less permissions.

---

I don't exactly understand why we would want to ship the original systemd service to be quite honest, since doing that would require doing a text replacement of the hardcoded path in the service file, where templating it with nix seems more sensible.

---

An example of why it is good to use upstream units when possible:

https://github.com/NixOS/nixpkgs/pull/147305

> I noticed that upstream has moved from a simple service to a notify service and we never caught that because we don't use the upstream service. Let's use the upstream unit so we don't miss things like that.

Note that you can patch the `.service` file in the package to contain the proper path: https://github.com/NixOS/nixpkgs/blob/97e0158a719f34e9170a67bdebaee956eb5bc61a/pkgs/servers/caddy/default.nix#L29

---

Fair point, I'll go ahead and implement this.